File: src/main/java/io/gittul/domain/github/GitHubRepositoryRepository.java
package io.gittul.domain.github;

import io.gittul.domain.github.entity.GitHubRepository;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface GitHubRepositoryRepository extends JpaRepository<GitHubRepository, Long> {
    Optional<GitHubRepository> findByRepoUrl(String repoUrl);
}

File: app/src/main/kotlin/io/gittul/app/domain/github/GitHubRepositoryRepository.kt
package io.gittul.app.domain.github

import io.gittul.core.domain.github.entity.GitHubRepository
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GitHubRepositoryRepository : JpaRepository<GitHubRepository, Long> {
    fun findByRepoUrl(repoUrl: String): Optional<GitHubRepository>
}
